conda_smithy/configure_feedstock.py

```diff
@@ -893,33 +908,39 @@ def _get_platforms_of_provider(provider, forge_config):
     keep_noarchs = []
     archs = []
     upload_packages = []
-    for platform in ["linux", "osx", "win"]:
-        for arch in ["64", "aarch64", "ppc64le", "armv7l"]:
-            platform_arch = (
-                platform if arch == "64" else "{}_{}".format(platform, arch)
-            )
-            if platform_arch not in forge_config["provider"]:
-                continue
-            if forge_config["provider"][platform_arch] == provider:
-                platforms.append(platform)
-                archs.append(arch)
-                if platform == "linux" and arch == "64":
-                    keep_noarchs.append(True)
-                else:
-                    keep_noarchs.append(False)
-                upload_packages.append(True)
-            elif (
-                provider == "azure"
-                and forge_config["azure"]["force"]
-                and arch == "64"
-            ):
-                platforms.append(platform)
-                archs.append(arch)
-                if platform == "linux" and arch == "64":
-                    keep_noarchs.append(True)
-                else:
-                    keep_noarchs.append(False)
-                upload_packages.append(False)
+    for platform_arch in forge_config["build_platform"].keys():
+        platform, arch = platform_arch.split("_")
+        build_platform_arch = forge_config["build_platform"][platform_arch]
+        build_platform, build_arch = build_platform_arch.split("_")
+        if (
+            build_arch == "64"
+            and build_platform in forge_config["provider"]
+            and forge_config["provider"][build_platform] is not None
+        ):
+            build_platform_arch = build_platform
+
+        if build_platform_arch not in forge_config["provider"]:
+            continue
+        if forge_config["provider"][build_platform_arch] == provider:
+            platforms.append(platform)
+            archs.append(arch)
+            if platform == "linux" and arch == "64":
+                keep_noarchs.append(True)
+            else:
+                keep_noarchs.append(False)
+            upload_packages.append(True)
+        elif (
+            provider == "azure"
+            and forge_config["azure"]["force"]
```

Comment:
Should the force condition be the if-clause, not the elif?

Reply:
Doesn't matter. We've had this for a while now.